### ISSUE TITLE
Compilation fail on module contents change

### ISSUE THREAD

**Issue by pvvrzv** [webdiscus/pug-plugin#77]
### **Describe the bug**
  Entry file `src/index.pug` has `script` element with `src` tag set to a javascript file which imports another javascript file with esm. Webpack compilation failes with an error **`Cannot read properties of undefined (reading 'length')`** when changes to the module javascript file were made.
  But if you make changes to entry javascript file after compilation fail webpack will recompile succesfully.

### **To Reproduce** 
  Clone [demo](https://github.com/pvvrzv/pug-plugin-compilation-fail-demo) repository, run `npm i`  and then `npm run server`. Make changes to `/src/module.js`.

### **Screenshots**
![error](https://github.com/webdiscus/pug-plugin/assets/79056000/020c4349-0cdc-4025-9a93-dc5bce2bb090)

### **Versions**

- node [18.16.0]
- pug-plugin [^4.9.7]
- webpack-cli [5.1.1]
- webpack-dev-server [4.15.0]
- webpack [5.83.1]

**Comment by webdiscus** [comment]
Hello @pvvrzv,

thanks for the issue report.
I try to fix it.

**Comment by webdiscus** [comment]
@pvvrzv 

I researched the problem. This is Webpack BUG, not in PugPlugin.
The Pug plugin do nothing with JS. All JS files are ignored in plugin because Webpack self handeln JS files.

### Workarounds

1. Use the `production` mode. The error occur only in `development` mode.
2. If you need `development` mode, then disable the caching:
```js
module.exports = {
  mode: 'development',
  cache: false, // <= fix issue
}
```

**Comment by pvvrzv** [comment]
@webdiscus

Thank you for your quick response.

Why exacty does this happen? Wepback by itself doesn't have any plroblem with caching.
I'm learning webpack internals currently, so it would be really helpful if you could point me in the right direction of this one.

By the way, do you use the plugin yourself? Is disabling caching the only option there is?

**Comment by webdiscus** [comment]
@pvvrzv 

Yes, I use the plugin myself. I created this plugin for many working projects to simplify Webpack config and to have many cool features that html-webpack-plugin + original pug-loader doesn't have.

Currently I don't see anything other than disabling caching for `development` mode when using ESM scripts. After changes Webpack try to rebuild  ESM modules, but not required top level script. I don't know why. In `production` mode everything works fine.

But I will research the problem in detail later, not now.

**Comment by webdiscus** [comment]
@pvvrzv 

The issue is fixed in the NEW release `v5.0.0` based on the [html-bundler-webpack-plugin](https://github.com/webdiscus/html-bundler-webpack-plugin).

You can try the [manual test](https://github.com/webdiscus/pug-plugin/tree/master/test/manual/watch-dependencies-import-js).

1. `npm start`
2. change the `src/module.js` file in the `serve` mode => no Webpack compilation error.
